examples/neighbors/plot_variables_choice.py

```diff
@@ -0,0 +1,79 @@
+"""
+=================================================================================
+Nearest Neighbors Classification with different values of n_neighbors and weights
+=================================================================================
+
+Sample usage of Nearest Neighbors classification.
+It will plot the decision boundaries for each class.
+
+This example show the effect of tweaking the parameters ``n_neighbors`` and ``weights``. 
+The smaller ``n_neighbors`` is the noisier the boundaries are.
```

Comment:
Could you please add paragraph to make the main take aways of this example explicit?

Here are some suggestions:

- for both weighting schemes, low values of ``n_neighbors`` lead to a very irregular decision boundary that is highly sensitive to the sampling noise of the training set and therefore lead it an overfitting predictor.
- for larger values of ``n_neighbors``, the decision boundary gets smoother and therefore less likely to overfit.
- with even large rvalues of ``n_neighobrs``, some points of the training set itself start to be mis-classified: the predictor is under-fit.
- for the extreme case where `n_neighbors >= n_samples == 150` with the uniform weighting scheme, all data points lie in the neighborhood of each the data point to classify and the resulting model is a constant predictor and is therefore pathologically under-fit.
- when using the the distance-based weighting scheme, this pathological under-fitting stays comparatively limited, because not all neighbors contribute equally to the decision function anymore.